build_tools/docker/cmake-emscripten/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# An image for building IREE through Emscripten using CMake.
+
+FROM gcr.io/iree-oss/cmake@sha256:9d9953acf5ca0cf1ff3e8de32f10f24dfab1c4e8ec5d1fc047f556024ee4bed6 AS final
+
+# See also
+#   * https://github.com/emscripten-core/emsdk/blob/main/docker/Dockerfile
+#   * https://hub.docker.com/r/emscripten/emsdk
+
+ARG EMSDK_COMMIT=5c0e31a03a72136ccaa761c408baf5a640f942ec
+ARG SDK_VERSION=2.0.25
+
+# Follow https://emscripten.org/docs/getting_started/downloads.html.
+RUN git clone https://github.com/emscripten-core/emsdk
+RUN cd emsdk && git checkout "${EMSDK_COMMIT?}" && \
+    ./emsdk install ${SDK_VERSION?} && \
+    ./emsdk activate ${SDK_VERSION?}
+
+# Set some environment variables for Emscripten to use.
+ENV EMSDK=/emsdk
+ENV EM_DATA=${EMSDK}/.data
+ENV EM_CONFIG=${EMSDK}/.emscripten
+ENV EM_CACHE=${EM_DATA}/cache
+ENV EM_PORTS=${EM_DATA}/ports
+# Emscripten writes into its cache location (outside of the CMake build
+# directory). We need to grant broad write permissions to this directory to be
+# able to run our scripts under different users.
+# See https://github.com/emscripten-core/emsdk/issues/535
+RUN mkdir -p ${EM_CACHE} && chmod -R 777 ${EM_CACHE}
```

Comment:
I think we discussed mounting a user home directory? That seems much nicer...

Reply:
We did, but I already had this working and didn't want to spend more time with Docker 😞
Could still figure that out, if you think it's worth it.

Reply:
Can you at least drop a note here that that's another option?